### PR TITLE
Fix coding problem: reassigning parameters #102

### LINKED ISSUE #102: Fix coding problem: reassigning parameters
image

### PR DESCRIPTION
**Pull request by jia1**
Fixes #102

### CODE REVIEW

File: src/seedu/addressbook/data/person/Email.java [hunk 1]
```diff
@@ -21,13 +21,13 @@
      *
      * @throws IllegalValueException if given email address string is invalid.
      */
-    public Email(String email, boolean isPrivate) throws IllegalValueException {
+    public Email(String emailValue, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        email = email.trim();
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Email.java]: This is the problematic line. i.e. changing the value of the parameter, which is a discouraged practice.
The fix is,
`String trimmedEmail = email.trim();`

Adding the `Value` suffix to the parameter doesn't solve the problem if you reassign it later. Furthermore, that additional suffix makes it inconsistent with other similar places.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -20,12 +20,12 @@
      *
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Address.java]: The PR name should contain the entire issue name, including the issue number. Just copy paste the whole thing. Your other PR had the same problem and I fixed it.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -20,12 +20,12 @@
...
      */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
+    public Address(String addressValue, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        if (!isValidAddress(addressValue)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        this.value = addressValue;
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Address.java]: This method doesn't have the mentioned problem.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 4]
```diff
@@ -45,7 +45,7 @@ public StorageOperationException(String message) {
 
     private final JAXBContext jaxbContext;
 
-    public final Path path;
+    public final Path filePath;
```
Comment (damithc) [inline src/seedu/addressbook/storage/StorageFile.java]: Is this related to this issue? If not, remove it from the PR.

**Comment by damithc** [comment]
Let us know when ready for review again.

**Comment by jia1** [comment]
Ready for review

**Comment by damithc** [comment]
Thanks for the contribution @jia1 
You may add yourself to the list of contributors in the README.md

**Pull request merged**